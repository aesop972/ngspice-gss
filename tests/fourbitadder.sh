#! /bin/sh

. $srcdir/config.sh
spicetest fourbitadder
