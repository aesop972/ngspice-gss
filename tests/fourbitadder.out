
Circuit: 4 bit adder


Initial Transient Solution
--------------------------

Node                                   Voltage
----                                   -------
99                                           5
1                                            0
2                                            0
3                                            0
4                                            0
5                                            0
6                                            0
7                                            0
8                                            0
1:1:1:1:9                            0.0179277
1:1:1:1:5                             0.826789
1:1:1:1:6                              4.97244
1:1:1:1:8                          3.54606e-09
1:1:1:1:7                              4.83208
1:1:1:1:10                             4.18989
1:1:1:7                                3.52783
1:1:1:2:9                            0.0363158
1:1:1:2:5                             0.844836
1:1:1:2:6                              4.99412
1:1:1:2:8                          3.66756e-09
1:1:1:2:7                              4.96415
1:1:1:2:10                             4.25286
1:1:1:8                                3.63074
1:1:1:3:9                            0.0363158
1:1:1:3:5                             0.844836
1:1:1:3:6                              4.99412
1:1:1:3:8                          3.66756e-09
1:1:1:3:7                              4.96415
1:1:1:3:10                             4.25286
1:1:1:9                                3.63074
1:1:1:4:9                              1.99534
1:1:1:4:5                              2.76452
1:1:1:4:6                              1.10792
1:1:1:4:8                              1.07019
1:1:1:4:7                              4.99984
1:1:1:4:10                            0.507271
1:1:1:10                             0.0253869
1:1:1:5:9                            0.0277172
1:1:1:5:5                             0.836398
1:1:1:5:6                              4.97251
1:1:1:5:8                          3.55613e-09
1:1:1:5:7                              4.83248
1:1:1:5:10                             4.19002
1:1:1:11                               3.52802
1:1:1:6:9                            0.0363158
1:1:1:6:5                             0.844836
1:1:1:6:6                               4.9941
1:1:1:6:8                          3.66742e-09
1:1:1:6:7                              4.96406
1:1:1:6:10                             4.25278
1:1:1:12                               3.63059
1:1:1:7:9                            0.0617017
1:1:1:7:5                             0.869755
1:1:1:7:6                               4.9941
1:1:1:7:8                           3.6935e-09
1:1:1:7:7                              4.96406
1:1:1:7:10                             4.25278
1:1:1:13                               3.63059
1:1:1:8:9                              1.98972
1:1:1:8:5                              2.75903
1:1:1:8:6                              1.10196
1:1:1:8:8                              1.06424
1:1:1:8:7                              4.99984
1:1:1:8:10                            0.500516
9                                    0.0178393
1:1:1:9:9                               1.9967
1:1:1:9:5                              2.76585
1:1:1:9:6                              1.10936
1:1:1:9:8                              1.07163
1:1:1:9:7                              4.99984
1:1:1:9:10                            0.508909
1:1:10                               0.0272212
1:1:2:1:9                            0.0179277
1:1:2:1:5                             0.826789
1:1:2:1:6                              4.97244
1:1:2:1:8                          3.54606e-09
1:1:2:1:7                              4.83208
1:1:2:1:10                             4.18989
1:1:2:7                                3.52783
1:1:2:2:9                            0.0363158
1:1:2:2:5                             0.844836
1:1:2:2:6                              4.99412
1:1:2:2:8                           3.6676e-09
1:1:2:2:7                              4.96417
1:1:2:2:10                             4.25288
1:1:2:8                                3.63077
1:1:2:3:9                            0.0363158
1:1:2:3:5                             0.844836
1:1:2:3:6                              4.99412
1:1:2:3:8                           3.6676e-09
1:1:2:3:7                              4.96417
1:1:2:3:10                             4.25288
1:1:2:9                                3.63077
1:1:2:4:9                               1.9967
1:1:2:4:5                              2.76585
1:1:2:4:6                              1.10936
1:1:2:4:8                              1.07163
1:1:2:4:7                              4.99984
1:1:2:4:10                            0.508909
1:1:2:10                             0.0272212
1:1:2:5:9                            0.0451489
1:1:2:5:5                              0.85351
1:1:2:5:6                              4.97258
1:1:2:5:8                          3.57407e-09
1:1:2:5:7                              4.83293
1:1:2:5:10                             4.19017
1:1:2:11                               3.52824
1:1:2:6:9                             0.063536
1:1:2:6:5                             0.871555
1:1:2:6:6                               4.9941
1:1:2:6:8                          3.69541e-09
1:1:2:6:7                              4.96406
1:1:2:6:10                             4.25278
1:1:2:12                               3.63059
1:1:2:7:9                             0.063536
1:1:2:7:5                             0.871555
1:1:2:7:6                               4.9941
1:1:2:7:8                          3.69541e-09
1:1:2:7:7                              4.96406
1:1:2:7:10                             4.25278
1:1:2:13                               3.63059
1:1:2:8:9                              1.98972
1:1:2:8:5                              2.75903
1:1:2:8:6                              1.10196
1:1:2:8:8                              1.06424
1:1:2:8:7                              4.99984
1:1:2:8:10                            0.500516
10                                   0.0178393
1:1:2:9:9                               1.9967
1:1:2:9:5                              2.76585
1:1:2:9:6                              1.10936
1:1:2:9:8                              1.07163
1:1:2:9:7                              4.99984
1:1:2:9:10                            0.508909
1:16                                 0.0272212
1:2:1:1:9                            0.0179277
1:2:1:1:5                             0.826789
1:2:1:1:6                              4.97244
1:2:1:1:8                          3.54606e-09
1:2:1:1:7                              4.83208
1:2:1:1:10                             4.18989
1:2:1:7                                3.52783
1:2:1:2:9                            0.0363158
1:2:1:2:5                             0.844836
1:2:1:2:6                              4.99412
1:2:1:2:8                           3.6676e-09
1:2:1:2:7                              4.96417
1:2:1:2:10                             4.25288
1:2:1:8                                3.63077
1:2:1:3:9                            0.0363158
1:2:1:3:5                             0.844836
1:2:1:3:6                              4.99412
1:2:1:3:8                           3.6676e-09
1:2:1:3:7                              4.96417
1:2:1:3:10                             4.25288
1:2:1:9                                3.63077
1:2:1:4:9                               1.9967
1:2:1:4:5                              2.76585
1:2:1:4:6                              1.10936
1:2:1:4:8                              1.07163
1:2:1:4:7                              4.99984
1:2:1:4:10                            0.508909
1:2:1:10                             0.0272212
1:2:1:5:9                            0.0451489
1:2:1:5:5                              0.85351
1:2:1:5:6                              4.97258
1:2:1:5:8                          3.57407e-09
1:2:1:5:7                              4.83293
1:2:1:5:10                             4.19017
1:2:1:11                               3.52824
1:2:1:6:9                             0.063536
1:2:1:6:5                             0.871555
1:2:1:6:6                               4.9941
1:2:1:6:8                          3.69541e-09
1:2:1:6:7                              4.96406
1:2:1:6:10                             4.25278
1:2:1:12                               3.63059
1:2:1:7:9                             0.063536
1:2:1:7:5                             0.871555
1:2:1:7:6                               4.9941
1:2:1:7:8                          3.69541e-09
1:2:1:7:7                              4.96406
1:2:1:7:10                             4.25278
1:2:1:13                               3.63059
1:2:1:8:9                              1.98972
1:2:1:8:5                              2.75903
1:2:1:8:6                              1.10196
1:2:1:8:8                              1.06424
1:2:1:8:7                              4.99984
1:2:1:8:10                            0.500516
11                                   0.0178393
1:2:1:9:9                               1.9967
1:2:1:9:5                              2.76585
1:2:1:9:6                              1.10936
1:2:1:9:8                              1.07163
1:2:1:9:7                              4.99984
1:2:1:9:10                            0.508909
1:2:10                               0.0272212
1:2:2:1:9                            0.0179277
1:2:2:1:5                             0.826789
1:2:2:1:6                              4.97243
1:2:2:1:8                          3.54601e-09
1:2:2:1:7                              4.83197
1:2:2:1:10                             4.18986
1:2:2:7                                3.52778
1:2:2:2:9                            0.0363158
1:2:2:2:5                             0.844836
1:2:2:2:6                              4.99412
1:2:2:2:8                           3.6676e-09
1:2:2:2:7                              4.96417
1:2:2:2:10                             4.25288
1:2:2:8                                3.63077
1:2:2:3:9                            0.0363158
1:2:2:3:5                             0.844836
1:2:2:3:6                              4.99412
1:2:2:3:8                           3.6676e-09
1:2:2:3:7                              4.96417
1:2:2:3:10                             4.25288
1:2:2:9                                3.63077
1:2:2:4:9                               1.9967
1:2:2:4:5                              2.76585
1:2:2:4:6                              1.10936
1:2:2:4:8                              1.07163
1:2:2:4:7                              4.99984
1:2:2:4:10                            0.508909
1:2:2:10                             0.0272212
1:2:2:5:9                            0.0451489
1:2:2:5:5                              0.85351
1:2:2:5:6                              4.97257
1:2:2:5:8                          3.57402e-09
1:2:2:5:7                              4.83282
1:2:2:5:10                             4.19014
1:2:2:11                               3.52819
1:2:2:6:9                             0.063536
1:2:2:6:5                             0.871555
1:2:2:6:6                               4.9941
1:2:2:6:8                          3.69541e-09
1:2:2:6:7                              4.96406
1:2:2:6:10                             4.25278
1:2:2:12                               3.63059
1:2:2:7:9                             0.063536
1:2:2:7:5                             0.871555
1:2:2:7:6                               4.9941
1:2:2:7:8                          3.69541e-09
1:2:2:7:7                              4.96406
1:2:2:7:10                             4.25278
1:2:2:13                               3.63059
1:2:2:8:9                              1.98972
1:2:2:8:5                              2.75903
1:2:2:8:6                              1.10196
1:2:2:8:8                              1.06424
1:2:2:8:7                              4.99984
1:2:2:8:10                            0.500516
12                                   0.0178393
1:2:2:9:9                              1.98972
1:2:2:9:5                              2.75903
1:2:2:9:6                              1.10196
1:2:2:9:8                              1.06424
1:2:2:9:7                              4.99984
1:2:2:9:10                            0.500516
13                                   0.0178393
vin4b#branch                        0.00207527
vin4a#branch                        0.00207527
vin3b#branch                        0.00207527
vin3a#branch                        0.00207527
vin2b#branch                        0.00207527
vin2a#branch                        0.00207527
vin1b#branch                        0.00207527
vin1a#branch                        0.00207527
vcc#branch                          -0.0757537

                                  4 bit adder
--------------------------------------------------------------------------------
Index   time            v(1)            
--------------------------------------------------------------------------------
0	0.000000e+00	0.000000e+00	
1	6.000000e-13	1.800000e-04	
2	7.422315e-13	2.226694e-04	
3	1.026694e-12	3.080083e-04	
4	1.595620e-12	4.786860e-04	
5	2.733472e-12	8.200415e-04	
6	5.009175e-12	1.502753e-03	
7	9.560581e-12	2.868174e-03	
8	1.866339e-11	5.599018e-03	
9	3.686902e-11	1.106071e-02	
10	7.328027e-11	2.198408e-02	
11	1.332803e-10	3.998408e-02	
12	1.932803e-10	5.798408e-02	
13	2.532803e-10	7.598408e-02	
14	3.132803e-10	9.398408e-02	
15	3.732803e-10	1.119841e-01	
16	4.332803e-10	1.299841e-01	
17	4.932803e-10	1.479841e-01	
18	5.532803e-10	1.659841e-01	
19	6.132803e-10	1.839841e-01	
20	6.732803e-10	2.019841e-01	
21	7.332803e-10	2.199841e-01	
22	7.932803e-10	2.379841e-01	
23	8.532803e-10	2.559841e-01	
24	9.132803e-10	2.739841e-01	
25	9.732803e-10	2.919841e-01	
26	1.033280e-09	3.099841e-01	
27	1.093280e-09	3.279841e-01	
28	1.153280e-09	3.459841e-01	
29	1.213280e-09	3.639841e-01	
30	1.273280e-09	3.819841e-01	
31	1.333280e-09	3.999841e-01	
32	1.393280e-09	4.179841e-01	
33	1.453280e-09	4.359841e-01	
34	1.513280e-09	4.539841e-01	
35	1.573280e-09	4.719841e-01	
36	1.633280e-09	4.899841e-01	
37	1.693280e-09	5.079841e-01	
38	1.753280e-09	5.259841e-01	
39	1.813280e-09	5.439841e-01	
40	1.873280e-09	5.619841e-01	
41	1.933280e-09	5.799841e-01	
42	1.993280e-09	5.979841e-01	
43	2.053280e-09	6.159841e-01	
44	2.113280e-09	6.339841e-01	
45	2.173280e-09	6.519841e-01	
46	2.233280e-09	6.699841e-01	
47	2.293280e-09	6.879841e-01	
48	2.353280e-09	7.059841e-01	
49	2.413280e-09	7.239841e-01	
50	2.473280e-09	7.419841e-01	
51	2.533280e-09	7.599841e-01	
52	2.593280e-09	7.779841e-01	
53	2.653280e-09	7.959841e-01	
54	2.713280e-09	8.139841e-01	

Index   time            v(1)            
--------------------------------------------------------------------------------
55	2.773280e-09	8.319841e-01	
56	2.833280e-09	8.499841e-01	
57	2.893280e-09	8.679841e-01	
58	2.953280e-09	8.859841e-01	
59	3.013280e-09	9.039841e-01	
60	3.073280e-09	9.219841e-01	
61	3.133280e-09	9.399841e-01	
62	3.193280e-09	9.579841e-01	
63	3.253280e-09	9.759841e-01	
64	3.313280e-09	9.939841e-01	
65	3.373280e-09	1.011984e+00	
66	3.433280e-09	1.029984e+00	
67	3.493280e-09	1.047984e+00	
68	3.553280e-09	1.065984e+00	
69	3.613280e-09	1.083984e+00	
70	3.673280e-09	1.101984e+00	
71	3.733280e-09	1.119984e+00	
72	3.793280e-09	1.137984e+00	
73	3.853280e-09	1.155984e+00	
74	3.913280e-09	1.173984e+00	
75	3.973280e-09	1.191984e+00	
76	4.033280e-09	1.209984e+00	
77	4.093280e-09	1.227984e+00	
78	4.153280e-09	1.245984e+00	
79	4.213280e-09	1.263984e+00	
80	4.273280e-09	1.281984e+00	
81	4.333280e-09	1.299984e+00	
82	4.393280e-09	1.317984e+00	
83	4.453280e-09	1.335984e+00	
84	4.513280e-09	1.353984e+00	
85	4.573280e-09	1.371984e+00	
86	4.633280e-09	1.389984e+00	
87	4.693280e-09	1.407984e+00	
88	4.753280e-09	1.425984e+00	
89	4.813280e-09	1.443984e+00	
90	4.873280e-09	1.461984e+00	
91	4.933280e-09	1.479984e+00	
92	4.993280e-09	1.497984e+00	
93	5.053280e-09	1.515984e+00	
94	5.113280e-09	1.533984e+00	
95	5.173280e-09	1.551984e+00	
96	5.233280e-09	1.569984e+00	
97	5.293280e-09	1.587984e+00	
98	5.353280e-09	1.605984e+00	
99	5.413280e-09	1.623984e+00	
100	5.473280e-09	1.641984e+00	
101	5.533280e-09	1.659984e+00	
102	5.593280e-09	1.677984e+00	
103	5.653280e-09	1.695984e+00	
104	5.713280e-09	1.713984e+00	
105	5.773280e-09	1.731984e+00	
106	5.833280e-09	1.749984e+00	
107	5.893280e-09	1.767984e+00	
108	5.953280e-09	1.785984e+00	
109	6.000000e-09	1.800000e+00	




